python/ray/autoscaler/v2/tests/test_sdk.py

```diff
@@ -460,6 +460,128 @@ def verify_cluster_no_node():
     wait_for_condition(verify_cluster_no_node)
 
 
+def test_idle_node_blocked(ray_start_cluster):
```

Comment:
nit: let's also write a small writeup for this test scenario. It's rather unique case.